sdk/python/feast/offline_server.py

```diff
@@ -0,0 +1,141 @@
+import ast
+import traceback
+from typing import Dict
+
+import pyarrow as pa
+import pyarrow.flight as fl
+
+from feast import FeatureStore
+
+
+class OfflineServer(fl.FlightServerBase):
+    def __init__(self, store: FeatureStore, location: str, **kwargs):
+        super(OfflineServer, self).__init__(location, **kwargs)
+        self._location = location
+        self.flights: Dict[str, Dict[str, str]] = {}
+        self.store = store
+
+    @classmethod
+    def descriptor_to_key(self, descriptor):
+        return (
+            descriptor.descriptor_type.value,
+            descriptor.command,
+            tuple(descriptor.path or tuple()),
+        )
+
+    # TODO: since we cannot anticipate here the call to get_historical_features call, what data should we return?
```

Comment:
This is a good point, I think we will have to extend `RetrievalJob` interface to expose schema of the result set in some way.

Reply:
Is this really relevant for our use case, BTW? our protocol uses `get_flight_info` only to retrieve the ticket before invoking the `do_get`,  it's not the regular use case for which the Arrow Flight protocol was initially designed for: there is no existing data set from which we can extract the schema and the metadata until we invoke the API on the store, and probably our client code is not interested in inspecting the descriptor.

Reply:
yup, probably not relevant right now if our client implementation can do without. Might be more important if we manage to polish API to a degree when it becomes usable even without feast client.

P.S. I wouldn't say it's not the regular use case for that particular reason. True, dataset isn't ready yet, but that can be said for any sql query to any database, right? Actually it's probably pretty trivial to extract schema without running the actual retrieval job, we know the schema for entity df and also know data types for all features.

Reply:
yep, schema is resolvable by means of the registry for the given features. we can use -1 for others requiring to run the query:
```
total_bytes: The size in bytes of the data in this flight, or -1 if unknown.
total_records: The total record count of this flight, or -1 if unknown.
```